Wikipedia/Code/WMFDatabaseHouseKeeper.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+@objc class WMFDatabaseHouseKeeper : NSObject {
+    
+    @objc func performHouseKeepingOnManagedObjectContext(_ moc: NSManagedObjectContext)  throws {
+        
+        try deleteStaleUnreferencedArticles(moc)
+    }
+    
+    private func deleteStaleUnreferencedArticles(_ moc: NSManagedObjectContext) throws {
+        
+        /**
+ 
+        Find `WMFContentGroup`s more than 30 days old.
+ 
+        */
+        
+        let midnightTodayUTC = (Date() as NSDate).wmf_midnightUTCDateFromLocal!
+        let utcCalendar = NSCalendar.wmf_utcGregorian() as Calendar
+        let thirtyDaysAgoMidnightUTC = utcCalendar.date(byAdding: .day, value: -30, to: midnightTodayUTC, wrappingComponents: true)
```

Comment:
Is `wrappingComponents: true` the better way to do this instead of `NSCalendarMatchStrictly`?